worlds/yoshisisland/__init__.py

```diff
@@ -185,12 +178,13 @@ def get_filler_item_name(self) -> str:
             return self.random.choice(filler_items)
 
     def set_rules(self) -> None:
-        if self.options.stage_logic == 0:
-            set_easy_rules(self)
-        elif self.options.stage_logic == 1:
-            set_normal_rules(self)
-        else:
-            set_hard_rules(self)
+        rules_per_difficulty = {
```

Comment:
I am like 95% sure this does not do what you think it does. Did you test it?

Reply:
It does! I have tested it and it is calling the proper function. I wasn't particularly happy with the multi-if statement